File: code/ex119.py
class Solution:
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        
        if rowIndex == 0:
            return [1]
        
        if rowIndex == 1:
            return [1, 1]


        res = [[1], [1, 1]]
        
        for i in range(2, rowIndex + 1):
            res.append([])

            for j in range(i + 1):
                if j == 0 or j == i:
                    res[i].append(1)
                else:
                    res[i].append(res[i - 1][j - 1] + res[i - 1][j])

        return res[rowIndex]
        

File: code/ex119.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> arr{1};
        for (int i = 1; i <= rowIndex; i++)
            getNext(arr);
        return arr;
    }
    
private:
    void getNext(vector<int>& arr) {
        int n = arr.size();
        for (int i = n - 1; i > 0; i--)
            arr[i] = arr[i] + arr[i - 1];
        arr.push_back(1);
    }
};